### PR TITLE
Add support for Cucumber-JVM 3 

### PR DESCRIPTION
**Pull request by noronhaa**
[//]: # (
. Thank you so much for sending us a pull request! 
.
. Make sure you have a clear name for your pull request. 
. The name should start with a capital letter and no dot is required in the end of the sentence.
. To link the request with isses use the following notation: (fixes #123, fixes #321\)
.
. An example of good pull request names:
. - Add Russian translation (fixes #123\)
. - Add an ability to disable default plugins
. - Support emoji in test descriptions
)

### Context
- Created a separate module for an Allure Cucumber-JVM 3 Listener (allure-cucumber3-jvm).
- Uses same classes from allure-cucumber2-jvm but has a cucumber-core and cucumber-java version 3.0 dependency instead.
- Where appropriate all depreciated or no longer existing methods (as of version 3.0) have been removed in favour of casting as outlined in the cucumber docs.
- Produces an identical report to using allure-cucumber2-jvm with Cucumber-JVM 2.

Usage:
- Add Dependency (only from local build at the moment)
```
    <dependency>
      <groupId>io.qameta.allure</groupId>
      <artifactId>allure-cucumber3-jvm</artifactId>
      <version>2.7-SNAPSHOT</version>
    </dependency>
```
- Add plugin to cucumber runner:
```
    plugin = {"io.qameta.allure.cucumber3jvm.AllureCucumber3Jvm"},
```
#### Checklist
- [x] [Sign Allure CLA][cla]
- [ ] Provide unit tests (same tests present as in allure-cucumber2-jvm for comparison)

[cla]: https://cla-assistant.io/accept/allure-framework/allure2

### CODE REVIEW

File: allure-cucumber3-jvm/src/main/java/io/qameta/allure/cucumber3jvm/AllureCucumber3Jvm.java [hunk 1]
```diff
@@ -0,0 +1,309 @@
+package io.qameta.allure.cucumber3jvm;
+
+import cucumber.api.*;
```
Comment (baev) [inline allure-cucumber3-jvm/src/main/java/io/qameta/allure/cucumber3jvm/AllureCucumber3Jvm.java]: do not use wildcard imports

File: allure-cucumber3-jvm/src/main/java/io/qameta/allure/cucumber3jvm/CucumberSourceUtils.java [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.IntStream;
+
+/**
+ * Parts of package-private cucumber.runtime.formatter.TestSourcesModel needed for Allure 2 adapter.
+ */
+public final class CucumberSourceUtils {
```
Comment (baev) [inline allure-cucumber3-jvm/src/main/java/io/qameta/allure/cucumber3jvm/CucumberSourceUtils.java]: lets make it package private or move to `internal` package

File: allure-cucumber3-jvm/src/test/resources/features/test-background.feature [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+  @good
+  Scenario: Scenario with background
+    When I add a to b
+    Then result is 15
+
+  @bad
+  Scenario: Bad scenario with background
+    When I add a to b
+    Then result is 16
```
Comment (baev) [inline allure-cucumber3-jvm/src/test/resources/features/test-background.feature]: make sure all the files ends with new line

**Comment by noronhaa** [comment]
Added requested changes

**Comment by baev** [comment]
@letsrokk could you please review this?

@noronhaa you may need to specify your git emails in Github profile in order to link the commits to it

**Comment by letsrokk** [comment]
@baev @noronhaa looks good to me in general

Although I do have suggestion for future improvement/refactoring - we need to think on how we can reduce code duplication here as I can see that some helper and utils classes were just copied for new adapter

Maybe we can introduce common shared module with such classes etc., but scope of this changes should be outside this PR. I'll have a look at it when we will have hands on new major Allure 3 version

**Comment by baev** [comment]
@noronhaa @letsrokk thanks!

**Pull request merged**